management/virtualmachine/client.go

```diff
@@ -99,105 +99,85 @@ func (self VirtualMachineClient) GetRole(cloudserviceName, deploymentName, roleN
-	requestId, azureErr := self.client.SendAzurePostRequest(requestURL, startRoleOperationBytes)
-	if azureErr != nil {
-		return azureErr
-	}
-
-	return self.client.WaitAsyncOperation(requestId)
+	return self.client.SendAzurePostRequest(requestURL, startRoleOperationBytes)
 }
 
-func (self VirtualMachineClient) ShutdownRole(cloudserviceName, deploymentName, roleName string) error {
+func (self VirtualMachineClient) ShutdownRole(cloudserviceName, deploymentName, roleName string) (string, error) {
 	if cloudserviceName == "" {
-		return fmt.Errorf(errParamNotSpecified, "cloudserviceName")
+		return "", fmt.Errorf(errParamNotSpecified, "cloudserviceName")
 	}
 	if deploymentName == "" {
-		return fmt.Errorf(errParamNotSpecified, "deploymentName")
+		return "", fmt.Errorf(errParamNotSpecified, "deploymentName")
 	}
 	if roleName == "" {
-		return fmt.Errorf(errParamNotSpecified, "roleName")
+		return "", fmt.Errorf(errParamNotSpecified, "roleName")
 	}
 
 	shutdownRoleOperationBytes, err := xml.Marshal(ShutdownRoleOperation{
 		OperationType: "ShutdownRoleOperation",
 	})
 	if err != nil {
-		return err
+		return "", err
 	}
 
 	requestURL := fmt.Sprintf(azureOperationsURL, cloudserviceName, deploymentName, roleName)
-	requestId, azureErr := self.client.SendAzurePostRequest(requestURL, shutdownRoleOperationBytes)
-	if azureErr != nil {
-		return azureErr
-	}
-
-	return self.client.WaitAsyncOperation(requestId)
+	return self.client.SendAzurePostRequest(requestURL, shutdownRoleOperationBytes)
 }
 
-func (self VirtualMachineClient) RestartRole(cloudserviceName, deploymentName, roleName string) error {
+func (self VirtualMachineClient) RestartRole(cloudserviceName, deploymentName, roleName string) (string, error) {
 	if cloudserviceName == "" {
-		return fmt.Errorf(errParamNotSpecified, "cloudserviceName")
+		return "", fmt.Errorf(errParamNotSpecified, "cloudserviceName")
 	}
 	if deploymentName == "" {
-		return fmt.Errorf(errParamNotSpecified, "deploymentName")
+		return "", fmt.Errorf(errParamNotSpecified, "deploymentName")
 	}
 	if roleName == "" {
-		return fmt.Errorf(errParamNotSpecified, "roleName")
+		return "", fmt.Errorf(errParamNotSpecified, "roleName")
 	}
 
 	restartRoleOperationBytes, err := xml.Marshal(RestartRoleOperation{
 		OperationType: "RestartRoleOperation",
 	})
 	if err != nil {
-		return err
+		return "", err
 	}
 
 	requestURL := fmt.Sprintf(azureOperationsURL, cloudserviceName, deploymentName, roleName)
-	requestId, azureErr := self.client.SendAzurePostRequest(requestURL, restartRoleOperationBytes)
-	if azureErr != nil {
-		return azureErr
-	}
-
-	return self.client.WaitAsyncOperation(requestId)
+	return self.client.SendAzurePostRequest(requestURL, restartRoleOperationBytes)
 }
 
-func (self VirtualMachineClient) DeleteRole(cloudserviceName, deploymentName, roleName string) error {
+func (self VirtualMachineClient) DeleteRole(cloudserviceName, deploymentName, roleName string) (string, error) {
```

Comment:
hmm shall we maybe introduce a type like `AzureOperationId` to make this obvious?